v8_c_api/src/v8_c_api.cpp

```diff
@@ -437,14 +454,31 @@ void v8_FreeContext(v8_context* ctx) {
 }
 
 void v8_SetPrivateData(v8_context* ctx, size_t index, void *pd) {
+	assert(pd);
+
 	v8::Local<v8::Context> v8_ctx = ctx->persistent_ctx->Get(ctx->isolate);
 	v8::Local<v8::External> data = v8::External::New(ctx->isolate, (void*)pd);
-	v8_ctx->SetEmbedderData(index + 2, data);
+	v8_ctx->SetEmbedderData(DATA_INDEX(index), data);
+}
+
+void v8_ResetPrivateData(v8_context *ctx, size_t index) {
+	v8::Local<v8::Context> v8_ctx = ctx->persistent_ctx->Get(ctx->isolate);
+	v8_ctx->SetEmbedderData(DATA_INDEX(index), v8::Null(ctx->isolate));
+}
+
+void v8_ResetPrivateDataOnCtxRef(v8_context_ref* ctx_ref, size_t index) {
+	ctx_ref->context->SetEmbedderData(DATA_INDEX(index), v8::Null(ctx_ref->context->GetIsolate()));
 }
 
 void* v8_GetPrivateData(v8_context* ctx, size_t index) {
 	v8::Local<v8::Context> v8_ctx = ctx->persistent_ctx->Get(ctx->isolate);
-	v8::Local<v8::External> data = v8::Local<v8::External>::Cast(v8_ctx->GetEmbedderData(index + 2));
+	v8::Local<v8::External> data = v8::Local<v8::External>::Cast(v8_ctx->GetEmbedderData(DATA_INDEX(index)));
```

Comment:
Can we cast `v8::Null` to `v8::External`?

Reply:
What I am asking is, assuming there will be V8::null on the slot of the `EmbedderData`, will our code crash in this case? Failing to cast it to `v8::External`?